site/gatsby-site/src/components/taxa/Taxonomy.js

```diff
@@ -0,0 +1,187 @@
+`;
+
+const renderTooltip = (props, displayText) => (
+  <Tooltip id="button-tooltip" {...props}>
+    {displayText}
+  </Tooltip>
+);
+
+const Taxonomy = ({ taxonomy, incidentId }) => {
+  const { isRole } = useUserContext();
+
+  const [showAllClassifications, setShowAllClassifications] = useState(false);
+
+  const [isEditing, setIsEditing] = useState(false);
+
+  const [showBanner, setShowBanner] = useState(false);
+
+  const canEdit =
+    isRole('taxonomy_editor') || isRole('taxonomy_editor_' + taxonomy.namespace.toLowerCase());
+
+  if (!canEdit && taxonomy.classificationsArray.length < 1) {
+    return <></>;
+  }
+
+  const handleSubmit = () => {
+    setIsEditing(false);
+    setShowBanner(true);
+  };
+
+  return (
+    <>
+      <TaxaCardHeader className="card-header">
+        <TaxaHeader>{`${taxonomy.namespace} Taxonomy Classifications`}</TaxaHeader>
+        <>
+          {isEditing ? (
+            <Button onClick={() => setIsEditing(false)}>Cancel</Button>
+          ) : (
+            canEdit && <Button onClick={() => setIsEditing(true)}>Edit</Button>
+          )}
+        </>
+        <a
+          style={{ order: 2, marginLeft: 'auto' }}
+          href={`/taxonomy/${taxonomy.namespace.toLowerCase()}`}
+        >
+          Taxonomy Details
+        </a>
+      </TaxaCardHeader>
+      <>
+        {!isEditing ? (
+          <>
+            {showBanner && (
+              <div style={{ padding: '0.5em' }}>
+                <Card bg="secondary" style={{ width: '100%' }} text="light" className="mb-2">
+                  <Card.Body>
+                    <Card.Text>
+                      Classifications will update in production within 24 hours.
+                    </Card.Text>
+                  </Card.Body>
+                </Card>
+              </div>
+            )}
+            {taxonomy.classificationsArray.length > 0 ? (
+              <>
+                {canEdit && (
+                  <ClassificationContainer key={'NOTES'} className="card-body">
+                    <Field>
+                      <OverlayTrigger
+                        placement="left"
+                        delay={{ show: 100, hide: 400 }}
+                        overlay={(e) => renderTooltip(e, 'Admin notes')}
+                      >
+                        <p>{'Notes'}</p>
+                      </OverlayTrigger>
+                    </Field>
+                    <Value>{taxonomy.notes}</Value>
+                  </ClassificationContainer>
+                )}
+                {taxonomy.classificationsArray
+                  .filter((field) => {
+                    if (showAllClassifications) return true;
+                    if (!showAllClassifications && field.weight >= 50) {
+                      return true;
+                    }
+                    return false;
+                  })
+                  .filter((field) => {
+                    if (field.name === 'Datasheets for Datasets' && field.value == 'No') {
```

Comment:
We are going to be adding 10s to 100s of the resources in the next few months. I don't think we can have these taxonomy-specific references sprinkled in the codebase. If some special logic is needed, it should likely be added to the taxa collection defining the specific taxonomy.

Reply:
Then we have to add another property that allows us to overwrite the way classifications are rendered.

Maybe adding an optional `render` property that if defined lets you set a different rendering method:

```
 {
         "short_name":"Datasheets for Datasets",
         "long_name":"Datasheets for Datasets",
         "short_description":"Document the",
         "long_description":"Lorem ipsum.",
         "display_type":"bool",
         "mongo_type":"bool",
         "default":null,
         "placeholder":null,
         "permitted_values":null,
         "weight":{
            "$numberInt":"70"
         },
         "instant_facet":true,
         "required":false         
         
        "render": "long_description" | "some other render method"
}
```
If omitted the actual value will be used, which is the current method.

What do you think?

Reply:
This is the intention of the `display_type` field.

Reply:
Yeah but this is a special case where the `display_type` is different when editing than when displaying the field:

If editing, it is presented as a yes/no select.
If displaying the value, the `long_description` field is shown.

Maybe use a `display_type: "long_description"`?